src/features/contracts/index.ts

```diff
@@ -92,24 +92,25 @@ const mapModel = async (
 		const contractValue =
 			_.get(contractEntry, mapper?.contractField) ?? mapper?.default;
 
-		if (mapper.refersTo) {
+		if (mapper.refersTo && contractValue != null) {
 			try {
-				const [arch] = await rootApi.get({
+				const [entry] = await rootApi.get({
 					resource: mapper.refersTo.resource,
 					options: {
 						$filter: { [mapper.refersTo.uniqueKey]: contractValue },
 						$select: ['id'],
 					},
 				});
 
-				mappedModel[key] = arch?.id;
+				mappedModel[key] = entry?.id;
```

Comment:
In this case `mappedModel[key]` will be `undefined` but below you've explicitly added a `?? null` to make sure it uses `null` rather than `undefined` - does this need the same?

Reply:
yes I think it does, cause Pine complains of the value is undefined, it expects null instead.